### DISCUSSION TITLE
Is there a way to exit or shutdown the backend server?

### DISCUSSION THREAD

**Discussion by jbrown123** [cpvalente/ontime#1215] [Q&A]
I'm running on windows. I use my machine for a variety of things, only sometimes for running shows. It would be nice not to have ontime occupying CPU and memory when it's not being used.

Selecting 'exit' from the windows interface just exits the UI but leaves the server running in the background. Is there a way to gracefully close the backend server?

**Comment by cpvalente** [comment]
> Selecting 'exit' from the windows interface just exits the UI but leaves the server running in the background. Is there a way to gracefully close the backend server?

Hi,

Are you running the app from the windows installer?

The "exit" button in the windows menu closes the window and leaves the app running in the background. This is to avoid accidental shutdowns of the server, especially in windows where closing the window closes the app

To shutdown the app you would want to navigate into settings in editor - settings - shutdown

Requesting shutdown here will shutdown the server process as you requested, with the exception of NPM and docker installs.

Can you please confirm this?

**Comment by jbrown123** [comment]
I installed the windows client on Win 10 64 bit. I launched the ontime UI / app from the windows start menu.

And yes, I can confirm that going to the gear icon and selecting "shutdown" from the settings page in the windows UI does shutdown the app, including the backend server.

I don't appear to be able to do this from a browser (localhost:4001/editor). The Shutdown button is greyed out. When I try to click the "shutdown ontime" button I get a "not" icon as my cursor.

![image](https://github.com/user-attachments/assets/fd57e63b-6763-4585-8ce6-65b7f916ec3a)

May I suggest that you add "how to shutdown" to the FAQ (https://docs.getontime.no/ontime/faq/)?

**Reply by cpvalente** [reply]
Thank you again for bringing this to my attention.

We disable the shutdown button when we see that the host name is not localhost. This is to prevent a user in another machine from shutting down Ontime. Ontime can only be shutdown from the machine it is running in

I wasn't aware that this was confusing for users and will add a task to improve the docs. Meanwhile, would you be available to make a suggestion of what the faq could say?

**Comment by jbrown123** [comment]
FYI, I was doing this ON the same machine, via localhost:4001/editor in Chrome. I was *UNABLE to shutdown* that way. Are there particular "users" (URLS) that are allowed to shutdown and others that are not? See this screenshot:

![image](https://github.com/user-attachments/assets/4d2372e1-8303-43cc-b50c-eb04def46389)

As far as the FAQ, I'd do something like this ...

## How do I shutdown the Ontime backend server?

You can *only* shutdown Ontime from the machine it is running on. This is a safety / security feature to prevent it being accidentally shutdown remotely.

To shutdown, go to the editor view (localhost:4001/editor) and click the gear icon in the upper left to open the settings. Then select "Shutdown" from the settings menu. Note the warning, then click the "Shutdown ontime" button. You will be prompted to confirm by clicking the "Shutdown" button in the dialog.

Note that on Windows you can do the same steps via the windows application / UI.

**Reply by cpvalente** [reply]
> FYI, I was doing this ON the same machine, via localhost:4001/editor in Chrome. I was *UNABLE to shutdown* that way. Are there particular "users" (URLS) that are allowed to shutdown and others that are not? See this screenshot:
> 
> 
> 
> ![image](https://github.com/user-attachments/assets/4d2372e1-8303-43cc-b50c-eb04def46389)
> 
> 
> 
> As far as the FAQ, I'd do something like this ...
> 
> 
> 
> ## How do I shutdown the Ontime backend server?
> 
> 
> 
> You can *only* shutdown Ontime from the machine it is running on. This is a safety / security feature to prevent it being accidentally shutdown remotely.
> 
> 
> 
> To shutdown, go to the editor view (localhost:4001/editor) and click the gear icon in the upper left to open the settings. Then select "Shutdown" from the settings menu. Note the warning, then click the "Shutdown ontime" button. You will be prompted to confirm by clicking the "Shutdown" button in the dialog.
> 
> 
> 
> Note that on Windows you can do the same steps via the windows application / UI.

Thank you for this
Just had a Quick Look at the code and it seems I have wrong information. It seems that you can only shutdown from the app window as you have experienced

We will discuss if it would make sense to expand this rule to allow localhost as well

**Reply by jbrown123** [reply]
Then, clearly, my FAQ entry needs to be tweaked to represent the actual rules.

How do you shutdown if you are running on Linux or Mac? Do you have a similar "app" for those platforms?

I'd suggest some sort of warning / indication / hint / something on the "shutdown" screen (http://localhost:4001/editor?settings=shutdown) about why the button is inactive. Clearly there is code that detects the situation because it's changing the interface. Have that same code add a note to the effect of "you must shutdown via the app on the machine running ontime" or similar.

**Reply by cpvalente** [reply]
Hi @jbrown123 , I have created changes on both the app and documentation to improve this.
Thank you again for bringing this to my attention
